base/docs/helpdb/Base.jl

```diff
@@ -1168,8 +1168,9 @@ Partially sort the vector `v` in place, according to the order specified by `by`
 `rev` so that the value at index `k` (or range of adjacent values if `k` is a range) occurs
 at the position where it would appear if the array were fully sorted via a non-stable
 algorithm. If `k` is a single index, that value is returned; if `k` is a range, an array of
-values at those indices is returned. Note that `select!` does not fully sort the input
-array.
+values at those indices is returned.
+
+Note that `select!` does not fully sort the input array, but it does ensure that all are correctly ordered with respect to the value at index `k`.
```

Comment:
Put this in a `!!! note` block, like in `statistics.jl`?

Also, maybe add a doctest example for the note?